libs/ui/src/hooks/use-activity.hook.ts

```diff
@@ -31,50 +32,109 @@ const transformApiData = (
   chainName: string
 ): SectionListActivityData[] => {
   let response: SectionListActivityData[] = [];
-  const filteredTransactions = data?.history_list.filter(
-    txData => !(txData.receives.length > 0 && txData.sends.length > 0) && txData.cate_id !== 'approve'
-  );
 
   let sectionListItem: SectionListActivityData | undefined;
 
   const userTokensMetadata: ActivityResponse['token_dict'] = data.token_dict;
+  const userProjectsMetadata: ActivityResponse['project_dict'] = data.project_dict;
 
-  filteredTransactions.forEach(txData => {
+  const getTokenSymbol = (address: string) =>
+    userTokensMetadata[address].optimized_symbol ?? userTokensMetadata[address].name;
+
+  const getProjectName = (projectId: string) => userProjectsMetadata[projectId].name;
+
+  const getTransactionType = (txData: TransactionResponse, isGasToken: boolean, isGasTokenSent: boolean) => {
+    if (txData.cate_id === TransactionTypeEnum.Send || (isGasToken && isGasTokenSent)) {
+      return {
+        type: TransactionTypeEnum.Send,
+        label: TransactionLabelEnum.Send
+      };
+    }
+    if (txData.cate_id === TransactionTypeEnum.Receive || (isGasToken && !isGasTokenSent)) {
+      return {
+        type: TransactionTypeEnum.Receive,
+        label: TransactionLabelEnum.Received
+      };
+    }
+    if (isNotEmptyString(txData.tx.name)) {
+      return {
+        type: TransactionTypeEnum.Interaction,
+        label: txData.tx.name
+      };
+    }
+
+    return {
+      type: TransactionTypeEnum.ContractInteraction,
+      label: TransactionLabelEnum.ContractInteraction
+    };
+  };
+
+  data?.history_list.forEach(txData => {
```

Comment:
try to omit using acronyms. rename `txData` to something more human-readable (`transaction` for example)